src/runtime/webgpu.cpp

```diff
@@ -0,0 +1,864 @@
+#include "HalideRuntimeWebGPU.h"
+#include "device_interface.h"
+#include "gpu_context_common.h"
+#include "printer.h"
+#include "scoped_spin_lock.h"
+
+// TODO: Discover this with CMake, and upstream toggle for stdint.h.
+#include "webgpu/webgpu.h"
```

Comment:
Code in src/runtime isn't allowed to include headers from the wider environment -- this is a hard requirement, to allow for cross-compiling on systems without having to have impossible SDK requirements everywhere. You'll need to pull out the forward declarations you need and put them here (see, e.g., `mini_opengl.h`). If the API for webgpu is too unstable for that to work, it's too unstable for Halide to use :-/

Reply:
OK. If this causes any unusual issues then let us know. (If it's just kinda icky-feeling, well, that's the case in all the other backends too, but as a practical matter, the stuff we assume is the stuff that isn't going to change cavalierly.)

Reply:
I've now done this. We'll need to update it every now and again to track changes to the native API while its still being developed, but this should eventually settle down.